TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/activities/SearchActivity.java

```diff
@@ -0,0 +1,167 @@
+                    Log.i(TAG, keyWord);
+                    performLocationSearch(keyWord);
+                    return true;
+                }
+                return false;
+            }
+        });
+
+        binding.restaurantSearch.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH) {
+                    if (selectedLocation == null) {
+                        binding.locationSearch.setError("Select a location first.");
+                        return true;
+                    }
+                    String keyWord = binding.restaurantSearch.getText().toString();
+                    Log.i(TAG, keyWord);
+                    performRestaurantSearch(selectedLocation, keyWord);
+                    return true;
+                }
+                return false;
+            }
+        });
+
+        binding.locationResultTextView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                binding.locationResultTextView.setVisibility(View.GONE);
+                binding.locationSearch.setVisibility(View.VISIBLE);
+                selectedLocation = null;
+            }
+        });
+    }
+
+    private void composeButtonListener(){
+        binding.composeFltButton.setVisibility(View.VISIBLE);
+        binding.composeFltButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                goToComposeFragment();
+            }
+        });
+    }
+
+    private void performLocationSearch(String keyWord) {
+        this.locationList = zomatoRequest.getLocations(keyWord);
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                Log.i(TAG, locationList.toString());
+                locationAdapter = new LocationAdapter(SearchActivity.this, locationList);
+                binding.resultsRecyclerView.setAdapter(locationAdapter);
+            }
+        }, 2000);
+    }
+
+    private void performRestaurantSearch(Location location, String keyWord) {
+        this.restaurantList = zomatoRequest.getRetaurants(location, keyWord, 0, 20);
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                Log.i(TAG, restaurantList.toString());
+                restaurantAdapter = new RestaurantAdapter(SearchActivity.this, restaurantList);
+                binding.resultsRecyclerView.setAdapter(restaurantAdapter);
+                selectedRestaurant = restaurantList.get(0);//set first result as default value
+                composeButtonListener();
+            }
+        }, 2000);
+    }
+
+    @Override
+    public void onClickLocation(int position) {
```

Comment:
Whenever you see a lot of code bunched up, this is usually a good spot to insert some whitespace to help with readability :-)